Bottom: 8098571666685aa6a8d637532398e0f42a184f64
Top:    ede00b8f02796535a32c5d52d9688cd8df285613
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-29 19:39:34 +0100

hid/gtk: Extend layer selector to allow editing layer names in situ


---

diff --git a/src/hid/gtk/ghid-layer-selector.c b/src/hid/gtk/ghid-layer-selector.c
index da96f57..432b443 100644
--- a/src/hid/gtk/ghid-layer-selector.c
+++ b/src/hid/gtk/ghid-layer-selector.c
@@ -40,6 +40,7 @@ enum {
   COLOR_COL,
   TEXT_COL,
   FONT_COL,
+  EDITABLE_COL,
   ACTIVATABLE_COL,
   SEPARATOR_COL,
   N_COLS
@@ -302,20 +303,31 @@ ghid_layer_selector_init (GHidLayerSelector *ls)
 
   renderer1 = ghid_cell_renderer_visibility_new ();
   renderer2 = gtk_cell_renderer_text_new ();
+  g_object_set (renderer2, "editable-set", TRUE, NULL);
 
-  opacity_col = gtk_tree_view_column_new_with_attributes ("", renderer1,
+  opacity_col = gtk_tree_view_column_new_with_attributes ("",
+                                                          renderer1,
                                                           "active", VISIBLE_COL,
-                                                          "color", COLOR_COL, NULL);
-  name_col = gtk_tree_view_column_new_with_attributes ("", renderer2,
+                                                          "color",  COLOR_COL,
+                                                          NULL);
+  name_col = gtk_tree_view_column_new_with_attributes ("",
+                                                       renderer2,
                                                        "text", TEXT_COL,
                                                        "font", FONT_COL,
+                                                       "editable", EDITABLE_COL,
                                                        NULL);
 
-  /* action index, active, color, text, font, is_separator */
-  ls->list_store = gtk_list_store_new (N_COLS, G_TYPE_POINTER, G_TYPE_INT,
-                                       G_TYPE_BOOLEAN, G_TYPE_STRING,
-                                       G_TYPE_STRING, G_TYPE_STRING,
-                                       G_TYPE_BOOLEAN, G_TYPE_BOOLEAN);
+  ls->list_store = gtk_list_store_new (N_COLS,
+                 /* STRUCT_COL      */ G_TYPE_POINTER,
+                 /* USER_ID_COL     */ G_TYPE_INT,
+                 /* VISIBLE_COL     */ G_TYPE_BOOLEAN,
+                 /* COLOR_COL       */ G_TYPE_STRING,
+                 /* TEXT_COL        */ G_TYPE_STRING,
+                 /* FONT_COL        */ G_TYPE_STRING,
+                 /* EDITABLE_COL    */ G_TYPE_BOOLEAN,
+                 /* ACTIVATABLE_COL */ G_TYPE_BOOLEAN,
+                 /* SEPARATOR_COL   */ G_TYPE_BOOLEAN);
+
   gtk_tree_view_insert_column (GTK_TREE_VIEW (ls), opacity_col, -1);
   gtk_tree_view_insert_column (GTK_TREE_VIEW (ls), name_col, -1);
   gtk_tree_view_set_model (GTK_TREE_VIEW (ls), GTK_TREE_MODEL (ls->list_store));
@@ -327,6 +339,7 @@ ghid_layer_selector_init (GHidLayerSelector *ls)
   ls->accel_group = gtk_accel_group_new ();
   ls->action_group = gtk_action_group_new ("LayerSelector");
   ls->n_actions = 0;
+
   for (i = 0; i < 20; ++i)
     ls->accel_available[i] = TRUE;
 
@@ -512,6 +525,7 @@ ghid_layer_selector_add_layer (GHidLayerSelector *ls,
                           COLOR_COL, color_string,
                           TEXT_COL, name,
                           FONT_COL, activatable ? NULL : "Italic",
+                          EDITABLE_COL, activatable,
                           ACTIVATABLE_COL, activatable,
                           SEPARATOR_COL, FALSE,
                           -1);
@@ -531,6 +545,7 @@ ghid_layer_selector_add_layer (GHidLayerSelector *ls,
                           COLOR_COL, color_string,
                           TEXT_COL, name,
                           FONT_COL, activatable ? NULL : "Italic",
+                          EDITABLE_COL, activatable,
                           ACTIVATABLE_COL, activatable,
                           -1);
     }
